### PR TITLE
Add more typing to the codebase and various fixes.

### PR DESCRIPTION
**Pull request by sodul**
- ensure the Makefile will behave more consistently across machines by setting the shell to Bash.
- update the coverage.py configuration in pyproject.toml to exclude `if TYPE_CHECKING:` blocks.
- fix potential bugs in the loader.
- fix default global kwargs dict in process.py.

### CODE REVIEW

**Comment by sodul** [comment]
The coverage reports are broken somehow, but only over GH Actions. Locally it seems to work fine either on macOS or through a Linux container:
```
> ./g -rv -s 9 ./green
...................................................................................................................s....................................................................................................................s

Skipped green.test.test_output.TestGreenStream.testDisableWindowsFalse - Colorama won't strip ANSI unless running on Windows

Skipped green.test.test_windows.TestWindows.test_colorOutput - This test is for windows-specific behavior.

Name                     Stmts   Miss  Cover   Missing
------------------------------------------------------
green/__init__.py            2      2     0%   1-3
green/__main__.py            3      3     0%   1-5
green/cmdline.py            59     12    80%   87-101
green/command.py            39      0   100%
green/config.py              2      2     0%   4-8
green/djangorunner.py       61      0   100%
green/examples.py           26     26     0%   1-68
green/exceptions.py          3      0   100%
green/junit.py             120      0   100%
green/loader.py            264      0   100%
green/output.py            101      0   100%
green/process.py            55      0   100%
green/result.py            405      1    99%   333
green/runner.py             73      0   100%
green/suite.py             122     13    89%   96, 100-103, 115-120, 126-129
green/test/__init__.py       0      0   100%
green/version.py             0      0   100%
------------------------------------------------------
TOTAL                     1335     59    96%

Ran 233 tests in 5.606s using 9 processes

OK (passes=231, skips=2)
```

I'll try to get that fixed as a separate issue.

File: green/loader.py [hunk 1]
```diff
@@ -153,13 +164,15 @@ def discover(self, current_path, file_pattern="test*.py", top_level_dir=None):
...
-        if type(targets) != list:
+    def loadTargets(
+        self, targets: Iterable[str] | str, file_pattern: str = "test*.py"
+    ) -> GreenTestSuite | None:
+        # If a string was passed in, put it into a tuple.
+        if isinstance(targets, str):
             targets = [targets]
 
         # Make sure there are no duplicate entries, preserving order
-        target_dict = OrderedDict()
```
Comment (IanFoertsch-clumio) [inline green/loader.py:162]: Ah, I see that [regular python dicts now have ordered dict functionality by default](https://docs.python.org/3/library/collections.html#ordereddict-objects).

**Review by IanFoertsch-clumio — APPROVED** [review]
👍 LGTM

File: green/loader.py (hunk 1 above)
Comment (sodul) [reply green/loader.py:162]: OrderedDict has more features but if they are not used, a regular dict is good enough.

**Pull request merged**